address.go

```diff
@@ -48,46 +55,48 @@ func newAddressGeneratorAtState(chainID ChainID, state addressState) *AddressGen
 	}
 }
 
-// NextAddress generates an account address and increments
-// the addressing state.
-//
-// The address is generated for a specific network (Flow mainnet, testnet, etc)
-// Each state is mapped to exactly one address. There are as many addresses
-// as states.
-//
-// zeroAddress() corresponds to the state "0" while ServiceAddress() corresponds to the
-// state "1".
-func (gen *AddressGenerator) NextAddress() (Address, error) {
-	err := gen.nextState()
-	if err != nil {
-		return zeroAddress(gen.chainID), err
-	}
+// Address generates an account address at the current addressing state.
+func (gen *AddressGenerator) Address() Address {
+	return generateAddress(gen.chainID, gen.state)
+}
 
-	return generateAddress(gen.chainID, gen.state), nil
+// NextAddress increments the addressing state and generates an account address.
+func (gen *AddressGenerator) NextAddress() Address {
+	gen.Next()
+	return generateAddress(gen.chainID, gen.state)
 }
 
-// nextState increments the addressing state.
+// Next increments the addressing state.
 //
-// The state values are incremented from 0 to 2^k-1
-func (gen *AddressGenerator) nextState() error {
+// State values are incremented from 0 to 2^k-1.
+func (gen *AddressGenerator) Next() *AddressGenerator {
 	if uint64(gen.state) > maxState {
-		return fmt.Errorf("the state value is not valid, it must be less or equal to %x", maxState)
+		panic(
+			fmt.Sprintf("addressing state must be less than or equal to %d", maxState),
+		)
 	}
-	gen.state += 1
-	return nil
+
+	gen.state++
+
+	return gen
+}
+
+// SetIndex fast-forwards or rewinds the addressing state to the given index.
+func (gen *AddressGenerator) SetIndex(i uint) *AddressGenerator {
```

Comment:
I think the idea of removing the error in `Next` assumes that it's unpractical to reach the index `2^45`, but with exposing this API, a caller can jump to that index and wouldn't get any error.  

Another comment: shouldn't a function `AddressAtIndex(i) Address` be one that a developer would use rather than `SetIndex(i) * AddressGenerator `. In both cases I feel like developers will use this API and skip all the others :)

Reply:
Anyway, I don't have a strong opinion about the API, you can set the interface you think is better. 
I didn't set it this way as 3 weeks ago the design of addresses didn't work with indices at all :)